### PR TITLE
Interaction System Messages

### PR DESCRIPTION
**Pull request by Acruid**
The InteractionSystem now raises EntitySystem events along with the existing interfaces.

### CODE REVIEW

**Comment by clusterfack** [comment]
Well there goes simple traceable logic flows, now we're going to use bound logic to create a side effect mutation in the message and checking it to return...???

And on an unrelated note, if everything else there is using two line if() return statements couldn't you just follow that standard?

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 1]
```diff
@@ -164,6 +161,13 @@ private void HandleUseItemInWorld(ICommonSession session, GridCoordinates coords
             if (!playerEnt.Transform.GridPosition.InRange(used.Transform.GridPosition, INTERACTION_RANGE))
                 return;
 
+            var activateMsg = new ActivateInWorldMessage(playerEnt, used);
+            RaiseEvent(activateMsg);
```
Comment (PJB3005) [inline Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs]: Events are raised in a deferred manner, so you'll never be able to catch this in time.

Comment (Acruid) [reply Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs]: Yeh, I forgot the other PR https://github.com/space-wizards/RobustToolbox/pull/794.

**Comment by Acruid** [comment]
> Well there goes simple traceable logic flows, now we're going to use bound logic to create a side effect mutation in the message and checking it to return...???
> 
> And on an unrelated note, if everything else there is using two line if() return statements couldn't you just follow that standard?

* See https://github.com/space-wizards/RobustToolbox/pull/794, it resolves the untraceable stack issue.
* Yes, I made mutable events.
* I pushed another commit fixing the formatting.

**Pull request merged**